static/js/components/EmploymentTab.js

```diff
@@ -65,56 +79,232 @@ class EmploymentTab extends ProfileTab {
     );
   }
 
-  employmentEntriesForm (index) {
-    let keySet = (key) => ['work_history', index, key];
+  componentWillMount () {
+    const { ui, setWorkHistoryEdit } = this.props;
+    if ( _.isUndefined(ui.workHistoryEdit) ) {
+      setWorkHistoryEdit(true);
+    }
+  }
+
+  toggleWorkHistoryEdit = () => {
+    const { ui, setWorkHistoryEdit } = this.props;
+    setWorkHistoryEdit(!ui.workHistoryEdit);
+  }
+
+  closeWorkDialog = () => {
+    const { setWorkDialogVisibility } = this.props;
+    setWorkDialogVisibility(false);
+  }
+
+  addWorkHistoryEntry = () => {
+    const {
+      updateProfile,
+      profile,
+      setWorkDialogIndex,
+      setWorkDialogVisibility,
+    } = this.props;
+    let clone = Object.assign({}, profile);
+    clone['work_history'] = clone['work_history'].concat(
+      _.cloneDeep(this.blankWorkHistoryEntry)
```

Comment:
cool, I just added a method to the `ProfilePage` class that we pass down to the tabs via `props`, and added a call to that method in the function to handle closing the dialog.